### ISSUE TITLE
Change OIDC default to Code + PKCE

### ISSUE THREAD

**Issue by Tratcher** [dotnet/aspnetcore#18932]
Code + PKCE is as secure as FormPost and avoids many of the issues with SameSite=None.

Change the nonce cookie SameSite default to lax? What about the correlation cookie?

**Comment by Tratcher** [comment]
Note this will also require the templates to provision a client secret. That's going to require tooling work to populate.

**Comment by blowdart** [comment]
As these templates are getting replaced, this would fall under the new templates, so closing.

**Comment by blowdart** [comment]
Auth handlers will need to switch too, based on the templates/tooling. Need to coordinate with @jmprieur

**Comment by blowdart** [comment]
@jmprieur in your tooling discussion don't you need this for IdentityWeb as you were needing a client secret to switch?

**Comment by Tratcher** [comment]
TODO: Test in chrome incognito mode which has started blocking SameSite cookies. We may have to change the cookie SameSite defaults to Unspecified.

**Comment by jmprieur** [comment]
Well @blowdart @Tratcher: MSAL.NET does not support Auth Code PKCE in confidential client apps. So if it's default, it needs to be disable-able when using AAD.

cc: @henrik-me FYI (and prioritization discussion)

**Comment by Tratcher** [comment]
We met with the Identity.Web team and decided to have them change the default in their library & templates first. In 6.0 we'll reconsider changing the default in OIDC.

**Comment by Tratcher** [comment]
I was finally able to test chrome by downloading Chromium. I enabled every current and future restriction available and the SameSite=none cookies continued to work. That makes this change safer because we can leave the cookies as is, but it also makes this change less necessary.

**Comment by halter73** [comment]
> MSAL.NET does not support Auth Code PKCE in confidential client apps.

@jmprieur Is this still an issue? My big worry is that changing the default after all this time is likely to break people even though authorization code flow is now widely supported. Some people are undoubtedly running outdated or custom stuff.

**Comment by jmprieur** [comment]
no, @halter73. Both MSAL.NET and Microsoft.Identity.Web support PKCE, and have for years
